### ISSUE TITLE
Finetuning in webUI broken

### ISSUE THREAD

**Issue by GamingDaveUk** [daswer123/xtts-webui#33]
New thread as its a different issue this time. 

So I load the webui, 
Click on train
In finetune model name I write commentator
I drop my audio sample ont the audio input section and wait for it to upload.
I click train.

I get this error:

```
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "H:\XTTS\xtts-webui\modules\train.py", line 239, in train_xtts_model
    shutil.copytree(str(ready_folder), str(
  File "C:\Program Files\Python310\lib\shutil.py", line 556, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'finetuned_models\\commentator\\ready'

```
The folder does not exist.

If I click train again it creates the folder and proceeds and I get this error:

finetuned_models\commentator
```
>> DVAE weights restored from: H:\XTTS\xtts-webui\models\v2.0.2\dvae.pth
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\modules\train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "H:\XTTS\xtts-webui\scripts\utils\gpt_train.py", line 175, in train_gpt
    train_samples, eval_samples = load_tts_samples(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\TTS\tts\datasets\__init__.py", line 121, in load_tts_samples
    assert len(meta_data_train) > 0, f" [!] No training samples found in {root_path}/{meta_file_train}"
AssertionError:  [!] No training samples found in finetuned_models\commentator\dataset/metadata_train.csv
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "H:\XTTS\xtts-webui\modules\train.py", line 245, in train_xtts_model
    shutil.copy(speaker_reference_path, reference_destination)
  File "C:\Program Files\Python310\lib\shutil.py", line 417, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "C:\Program Files\Python310\lib\shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'finetuned_models\\commentator\\ready\\reference.wav'
```
If i place a wav file called reference.wav in the folder I can proceed to the next error. 

```
Existing language matches target language
Loading Whisper Model!
Existing training metadata found and loaded.
Existing evaluation metadata found and loaded.
Dataset Processed!
finetuned_models\commentator
>> DVAE weights restored from: H:\XTTS\xtts-webui\models\v2.0.2\dvae.pth
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\modules\train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "H:\XTTS\xtts-webui\scripts\utils\gpt_train.py", line 175, in train_gpt
    train_samples, eval_samples = load_tts_samples(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\TTS\tts\datasets\__init__.py", line 121, in load_tts_samples
    assert len(meta_data_train) > 0, f" [!] No training samples found in {root_path}/{meta_file_train}"
AssertionError:  [!] No training samples found in finetuned_models\commentator\dataset/metadata_train.csv
```

No idea past this point. 

Metadata_train.csv just includes "audio_file|text|speaker_name" 

Full log including the Git pull, installer run (to make sure requirements are installed) and all the errors:

```
Microsoft Windows [Version 10.0.22621.3007]
(c) Microsoft Corporation. All rights reserved.

H:\XTTS\xtts-webui>git pull
Already up to date.

H:\XTTS\xtts-webui>install.bat
Requirement already satisfied: gradio==4.13.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 1)) (4.13.0)
Requirement already satisfied: torch==2.1.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 2)) (2.1.1+cu118)
Requirement already satisfied: torchaudio==2.1.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 3)) (2.1.1+cu118)
Requirement already satisfied: faster_whisper==0.10.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 4)) (0.10.0)
Requirement already satisfied: tts>=0.22.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 5)) (0.22.0)
Requirement already satisfied: langid in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 6)) (1.1.6)
Requirement already satisfied: noisereduce in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 7)) (3.0.0)
Requirement already satisfied: pedalboard in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 8)) (0.8.7)
Requirement already satisfied: pydub in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 9)) (0.25.1)
Requirement already satisfied: ffmpeg-python in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 10)) (0.2.0)
Requirement already satisfied: soundfile in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 11)) (0.12.1)
Requirement already satisfied: cutlet in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 12)) (0.3.0)
Requirement already satisfied: fugashi[unidic-lite] in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 13)) (1.3.0)
Requirement already satisfied: loguru in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 14)) (0.7.2)
Requirement already satisfied: omegaconf==2.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 15)) (2.3.0)
Requirement already satisfied: resampy==0.4.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 16)) (0.4.2)
Requirement already satisfied: tabulate==0.8.10 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 17)) (0.8.10)
Requirement already satisfied: requests in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 18)) (2.31.0)
Requirement already satisfied: faiss-cpu in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 19)) (1.7.4)
Requirement already satisfied: pyworld in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 20)) (0.3.4)
Requirement already satisfied: torchcrepe in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 21)) (0.0.22)
Requirement already satisfied: praat-parselmouth>=0.4.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 22)) (0.4.3)
Requirement already satisfied: translators in h:\xtts\xtts-webui\venv\lib\site-packages (from -r .\requirements.txt (line 23)) (5.8.9)
Requirement already satisfied: huggingface-hub>=0.19.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.20.2)
Requirement already satisfied: pillow<11.0,>=8.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (10.2.0)
Requirement already satisfied: ffmpy in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.3.1)
Requirement already satisfied: importlib-resources<7.0,>=1.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (6.1.1)
Requirement already satisfied: aiofiles<24.0,>=22.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (23.2.1)
Requirement already satisfied: pydantic>=2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (2.5.3)
Requirement already satisfied: jinja2<4.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (3.1.2)
Requirement already satisfied: numpy~=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (1.22.0)
Requirement already satisfied: fastapi in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.108.0)
Requirement already satisfied: tomlkit==0.12.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.12.0)
Requirement already satisfied: pandas<3.0,>=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (1.5.3)
Requirement already satisfied: httpx in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.26.0)
Requirement already satisfied: uvicorn>=0.14.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.25.0)
Requirement already satisfied: altair<6.0,>=4.2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (5.2.0)
Requirement already satisfied: gradio-client==0.8.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.8.0)
Requirement already satisfied: semantic-version~=2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (2.10.0)
Requirement already satisfied: typing-extensions~=4.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (4.9.0)
Requirement already satisfied: packaging in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (23.2)
Requirement already satisfied: matplotlib~=3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (3.8.2)
Requirement already satisfied: python-multipart in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.0.6)
Requirement already satisfied: orjson~=3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (3.9.10)
Requirement already satisfied: markupsafe~=2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (2.1.3)
Requirement already satisfied: pyyaml<7.0,>=5.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (6.0.1)
Requirement already satisfied: typer[all]<1.0,>=0.9 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio==4.13.0->-r .\requirements.txt (line 1)) (0.9.0)
Requirement already satisfied: sympy in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1->-r .\requirements.txt (line 2)) (1.12)
Requirement already satisfied: fsspec in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1->-r .\requirements.txt (line 2)) (2023.12.2)
Requirement already satisfied: filelock in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1->-r .\requirements.txt (line 2)) (3.13.1)
Requirement already satisfied: networkx in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1->-r .\requirements.txt (line 2)) (2.8.8)
Requirement already satisfied: av==10.* in h:\xtts\xtts-webui\venv\lib\site-packages (from faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (10.0.0)
Requirement already satisfied: ctranslate2<4,>=3.22 in h:\xtts\xtts-webui\venv\lib\site-packages (from faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (3.23.0)
Requirement already satisfied: tokenizers<0.16,>=0.13 in h:\xtts\xtts-webui\venv\lib\site-packages (from faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (0.15.0)
Requirement already satisfied: onnxruntime<2,>=1.14 in h:\xtts\xtts-webui\venv\lib\site-packages (from faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (1.16.3)
Requirement already satisfied: antlr4-python3-runtime==4.9.* in h:\xtts\xtts-webui\venv\lib\site-packages (from omegaconf==2.3.0->-r .\requirements.txt (line 15)) (4.9.3)
Requirement already satisfied: numba>=0.53 in h:\xtts\xtts-webui\venv\lib\site-packages (from resampy==0.4.2->-r .\requirements.txt (line 16)) (0.58.1)
Requirement already satisfied: websockets<12.0,>=10.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gradio-client==0.8.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (11.0.3)
Requirement already satisfied: jieba in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.42.1)
Requirement already satisfied: bnunicodenormalizer in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.1.6)
Requirement already satisfied: encodec>=0.1.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.1.1)
Requirement already satisfied: bangla in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.0.2)
Requirement already satisfied: bnnumerizer in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.0.2)
Requirement already satisfied: cython>=0.29.30 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.7)
Requirement already satisfied: aiohttp>=3.8.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (3.9.1)
Requirement already satisfied: transformers>=4.33.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (4.36.2)
Requirement already satisfied: unidecode>=1.3.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (1.3.7)
Requirement already satisfied: nltk in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (3.8.1)
Requirement already satisfied: einops>=0.6.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.7.0)
Requirement already satisfied: librosa>=0.10.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.10.0)
Requirement already satisfied: trainer>=0.0.32 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.0.36)
Requirement already satisfied: jamo in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.4.1)
Requirement already satisfied: g2pkk>=0.1.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.1.2)
Requirement already satisfied: spacy[ja]>=3 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (3.7.2)
Requirement already satisfied: scipy>=1.11.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (1.11.4)
Requirement already satisfied: umap-learn>=0.5.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.5.5)
Requirement already satisfied: flask>=2.0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.0)
Requirement already satisfied: hangul-romanize in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.1.0)
Requirement already satisfied: gruut[de,es,fr]==2.2.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (2.2.3)
Requirement already satisfied: tqdm>=4.64.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (4.66.1)
Requirement already satisfied: coqpit>=0.0.16 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.0.17)
Requirement already satisfied: pysbd>=0.3.4 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3.4)
Requirement already satisfied: pypinyin in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.50.0)
Requirement already satisfied: num2words in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.5.13)
Requirement already satisfied: scikit-learn>=1.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (1.3.2)
Requirement already satisfied: anyascii>=0.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3.2)
Requirement already satisfied: inflect>=5.6.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tts>=0.22.0->-r .\requirements.txt (line 5)) (7.0.0)
Requirement already satisfied: Babel<3.0.0,>=2.8.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.14.0)
Requirement already satisfied: dateparser~=1.1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.1.8)
Requirement already satisfied: gruut-ipa<1.0,>=0.12.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.13.0)
Requirement already satisfied: gruut_lang_en~=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.0)
Requirement already satisfied: jsonlines~=1.2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.2.0)
Requirement already satisfied: python-crfsuite~=0.9.7 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.9.10)
Requirement already satisfied: gruut_lang_fr~=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.2)
Requirement already satisfied: gruut_lang_es~=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.0)
Requirement already satisfied: gruut_lang_de~=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.0)
Requirement already satisfied: future in h:\xtts\xtts-webui\venv\lib\site-packages (from ffmpeg-python->-r .\requirements.txt (line 10)) (0.18.3)
Requirement already satisfied: cffi>=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from soundfile->-r .\requirements.txt (line 11)) (1.16.0)
Requirement already satisfied: jaconv in h:\xtts\xtts-webui\venv\lib\site-packages (from cutlet->-r .\requirements.txt (line 12)) (0.3.4)
Requirement already satisfied: mojimoji in h:\xtts\xtts-webui\venv\lib\site-packages (from cutlet->-r .\requirements.txt (line 12)) (0.0.12)
Requirement already satisfied: unidic-lite in h:\xtts\xtts-webui\venv\lib\site-packages (from fugashi[unidic-lite]->-r .\requirements.txt (line 13)) (1.0.8)
Requirement already satisfied: win32-setctime>=1.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from loguru->-r .\requirements.txt (line 14)) (1.1.0)
Requirement already satisfied: colorama>=0.3.4 in h:\xtts\xtts-webui\venv\lib\site-packages (from loguru->-r .\requirements.txt (line 14)) (0.4.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from requests->-r .\requirements.txt (line 18)) (2.1.0)
Requirement already satisfied: idna<4,>=2.5 in h:\xtts\xtts-webui\venv\lib\site-packages (from requests->-r .\requirements.txt (line 18)) (3.6)
Requirement already satisfied: charset-normalizer<4,>=2 in h:\xtts\xtts-webui\venv\lib\site-packages (from requests->-r .\requirements.txt (line 18)) (3.3.2)
Requirement already satisfied: certifi>=2017.4.17 in h:\xtts\xtts-webui\venv\lib\site-packages (from requests->-r .\requirements.txt (line 18)) (2023.11.17)
Requirement already satisfied: pathos>=0.2.9 in h:\xtts\xtts-webui\venv\lib\site-packages (from translators->-r .\requirements.txt (line 23)) (0.3.1)
Requirement already satisfied: cryptography>=38.0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from translators->-r .\requirements.txt (line 23)) (41.0.7)
Requirement already satisfied: PyExecJS>=1.5.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from translators->-r .\requirements.txt (line 23)) (1.5.1)
Requirement already satisfied: lxml>=4.9.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from translators->-r .\requirements.txt (line 23)) (5.1.0)
Requirement already satisfied: multidict<7.0,>=4.5 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (6.0.4)
Requirement already satisfied: aiosignal>=1.1.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.3.1)
Requirement already satisfied: frozenlist>=1.1.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.4.1)
Requirement already satisfied: async-timeout<5.0,>=4.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (4.0.3)
Requirement already satisfied: attrs>=17.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (23.2.0)
Requirement already satisfied: yarl<2.0,>=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from aiohttp>=3.8.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.9.4)
Requirement already satisfied: jsonschema>=3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from altair<6.0,>=4.2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (4.20.0)
Requirement already satisfied: toolz in h:\xtts\xtts-webui\venv\lib\site-packages (from altair<6.0,>=4.2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.12.0)
Requirement already satisfied: pycparser in h:\xtts\xtts-webui\venv\lib\site-packages (from cffi>=1.0->soundfile->-r .\requirements.txt (line 11)) (2.21)
Requirement already satisfied: setuptools in h:\xtts\xtts-webui\venv\lib\site-packages (from ctranslate2<4,>=3.22->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (63.2.0)
Requirement already satisfied: Werkzeug>=3.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from flask>=2.0.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.1)
Requirement already satisfied: blinker>=1.6.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from flask>=2.0.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.7.0)
Requirement already satisfied: click>=8.1.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from flask>=2.0.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (8.1.7)
Requirement already satisfied: itsdangerous>=2.1.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from flask>=2.0.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.1.2)
Requirement already satisfied: lazy-loader>=0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3)
Requirement already satisfied: audioread>=2.1.9 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.1)
Requirement already satisfied: msgpack>=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.0.7)
Requirement already satisfied: decorator>=4.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (5.1.1)
Requirement already satisfied: soxr>=0.3.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3.7)
Requirement already satisfied: joblib>=0.14 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.3.2)
Requirement already satisfied: pooch>=1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.8.0)
Requirement already satisfied: fonttools>=4.22.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (4.47.0)
Requirement already satisfied: pyparsing>=2.3.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (3.1.1)
Requirement already satisfied: cycler>=0.10 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.12.1)
Requirement already satisfied: python-dateutil>=2.7 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (2.8.2)
Requirement already satisfied: kiwisolver>=1.3.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.4.5)
Requirement already satisfied: contourpy>=1.0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from matplotlib~=3.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.2.0)
Requirement already satisfied: docopt>=0.6.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from num2words->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.6.2)
Requirement already satisfied: llvmlite<0.42,>=0.41.0dev0 in h:\xtts\xtts-webui\venv\lib\site-packages (from numba>=0.53->resampy==0.4.2->-r .\requirements.txt (line 16)) (0.41.1)
Requirement already satisfied: flatbuffers in h:\xtts\xtts-webui\venv\lib\site-packages (from onnxruntime<2,>=1.14->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (23.5.26)
Requirement already satisfied: coloredlogs in h:\xtts\xtts-webui\venv\lib\site-packages (from onnxruntime<2,>=1.14->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (15.0.1)
Requirement already satisfied: protobuf in h:\xtts\xtts-webui\venv\lib\site-packages (from onnxruntime<2,>=1.14->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (4.23.4)
Requirement already satisfied: pytz>=2020.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from pandas<3.0,>=1.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (2023.3.post1)
Requirement already satisfied: pox>=0.3.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from pathos>=0.2.9->translators->-r .\requirements.txt (line 23)) (0.3.3)
Requirement already satisfied: ppft>=1.7.6.7 in h:\xtts\xtts-webui\venv\lib\site-packages (from pathos>=0.2.9->translators->-r .\requirements.txt (line 23)) (1.7.6.7)
Requirement already satisfied: dill>=0.3.7 in h:\xtts\xtts-webui\venv\lib\site-packages (from pathos>=0.2.9->translators->-r .\requirements.txt (line 23)) (0.3.7)
Requirement already satisfied: multiprocess>=0.70.15 in h:\xtts\xtts-webui\venv\lib\site-packages (from pathos>=0.2.9->translators->-r .\requirements.txt (line 23)) (0.70.15)
Requirement already satisfied: pydantic-core==2.14.6 in h:\xtts\xtts-webui\venv\lib\site-packages (from pydantic>=2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (2.14.6)
Requirement already satisfied: annotated-types>=0.4.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from pydantic>=2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.6.0)
Requirement already satisfied: six>=1.10.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from PyExecJS>=1.5.1->translators->-r .\requirements.txt (line 23)) (1.16.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from scikit-learn>=1.3.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.2.0)
Requirement already satisfied: smart-open<7.0.0,>=5.2.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (6.4.0)
Requirement already satisfied: langcodes<4.0.0,>=3.2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.3.0)
Requirement already satisfied: wasabi<1.2.0,>=0.9.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.1.2)
Requirement already satisfied: spacy-legacy<3.1.0,>=3.0.11 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.12)
Requirement already satisfied: weasel<0.4.0,>=0.1.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3.4)
Requirement already satisfied: catalogue<2.1.0,>=2.0.6 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.10)
Requirement already satisfied: cymem<2.1.0,>=2.0.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.8)
Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.0.10)
Requirement already satisfied: srsly<3.0.0,>=2.4.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.4.8)
Requirement already satisfied: preshed<3.1.0,>=3.0.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.0.9)
Requirement already satisfied: spacy-loggers<2.0.0,>=1.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.0.5)
Requirement already satisfied: thinc<8.3.0,>=8.1.8 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (8.2.2)
Requirement already satisfied: sudachidict-core>=20211220 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (20230927)
Requirement already satisfied: sudachipy!=0.6.1,>=0.5.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.6.8)
Requirement already satisfied: tensorboard in h:\xtts\xtts-webui\venv\lib\site-packages (from trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.15.1)
Requirement already satisfied: psutil in h:\xtts\xtts-webui\venv\lib\site-packages (from trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (5.9.7)
Requirement already satisfied: safetensors>=0.3.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from transformers>=4.33.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.4.1)
Requirement already satisfied: regex!=2019.12.17 in h:\xtts\xtts-webui\venv\lib\site-packages (from transformers>=4.33.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (2023.12.25)
Requirement already satisfied: rich<14.0.0,>=10.11.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from typer[all]<1.0,>=0.9->gradio==4.13.0->-r .\requirements.txt (line 1)) (13.7.0)
Requirement already satisfied: shellingham<2.0.0,>=1.3.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from typer[all]<1.0,>=0.9->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.5.4)
Requirement already satisfied: pynndescent>=0.5 in h:\xtts\xtts-webui\venv\lib\site-packages (from umap-learn>=0.5.1->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.5.11)
Requirement already satisfied: h11>=0.8 in h:\xtts\xtts-webui\venv\lib\site-packages (from uvicorn>=0.14.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.14.0)
Requirement already satisfied: starlette<0.33.0,>=0.29.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from fastapi->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.32.0.post1)
Requirement already satisfied: anyio in h:\xtts\xtts-webui\venv\lib\site-packages (from httpx->gradio==4.13.0->-r .\requirements.txt (line 1)) (4.2.0)
Requirement already satisfied: httpcore==1.* in h:\xtts\xtts-webui\venv\lib\site-packages (from httpx->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.0.2)
Requirement already satisfied: sniffio in h:\xtts\xtts-webui\venv\lib\site-packages (from httpx->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.3.0)
Requirement already satisfied: mpmath>=0.19 in h:\xtts\xtts-webui\venv\lib\site-packages (from sympy->torch==2.1.1->-r .\requirements.txt (line 2)) (1.3.0)
Requirement already satisfied: tzlocal in h:\xtts\xtts-webui\venv\lib\site-packages (from dateparser~=1.1.0->gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (5.2)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in h:\xtts\xtts-webui\venv\lib\site-packages (from jsonschema>=3.0->altair<6.0,>=4.2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (2023.12.1)
Requirement already satisfied: referencing>=0.28.4 in h:\xtts\xtts-webui\venv\lib\site-packages (from jsonschema>=3.0->altair<6.0,>=4.2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.32.1)
Requirement already satisfied: rpds-py>=0.7.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from jsonschema>=3.0->altair<6.0,>=4.2.0->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.16.2)
Requirement already satisfied: platformdirs>=2.5.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from pooch>=1.0->librosa>=0.10.0->tts>=0.22.0->-r .\requirements.txt (line 5)) (4.1.0)
Requirement already satisfied: markdown-it-py>=2.2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from rich<14.0.0,>=10.11.0->typer[all]<1.0,>=0.9->gradio==4.13.0->-r .\requirements.txt (line 1)) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from rich<14.0.0,>=10.11.0->typer[all]<1.0,>=0.9->gradio==4.13.0->-r .\requirements.txt (line 1)) (2.17.2)
Requirement already satisfied: exceptiongroup>=1.0.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from anyio->httpx->gradio==4.13.0->-r .\requirements.txt (line 1)) (1.2.0)
Requirement already satisfied: blis<0.8.0,>=0.7.8 in h:\xtts\xtts-webui\venv\lib\site-packages (from thinc<8.3.0,>=8.1.8->spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.7.11)
Requirement already satisfied: confection<1.0.0,>=0.0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from thinc<8.3.0,>=8.1.8->spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.1.4)
Requirement already satisfied: cloudpathlib<0.17.0,>=0.7.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from weasel<0.4.0,>=0.1.0->spacy[ja]>=3->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.16.0)
Requirement already satisfied: humanfriendly>=9.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from coloredlogs->onnxruntime<2,>=1.14->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (10.0)
Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.7.2)
Requirement already satisfied: markdown>=2.6.8 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.5.1)
Requirement already satisfied: google-auth<3,>=1.6.3 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.26.1)
Requirement already satisfied: absl-py>=0.4 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (2.0.0)
Requirement already satisfied: grpcio>=1.48.2 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.60.0)
Requirement already satisfied: google-auth-oauthlib<2,>=0.5 in h:\xtts\xtts-webui\venv\lib\site-packages (from tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.2.0)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (5.3.2)
Requirement already satisfied: pyasn1-modules>=0.2.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.3.0)
Requirement already satisfied: rsa<5,>=3.1.4 in h:\xtts\xtts-webui\venv\lib\site-packages (from google-auth<3,>=1.6.3->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (4.9)
Requirement already satisfied: requests-oauthlib>=0.7.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from google-auth-oauthlib<2,>=0.5->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (1.3.1)
Requirement already satisfied: pyreadline3 in h:\xtts\xtts-webui\venv\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime<2,>=1.14->faster_whisper==0.10.0->-r .\requirements.txt (line 4)) (3.4.1)
Requirement already satisfied: mdurl~=0.1 in h:\xtts\xtts-webui\venv\lib\site-packages (from markdown-it-py>=2.2.0->rich<14.0.0,>=10.11.0->typer[all]<1.0,>=0.9->gradio==4.13.0->-r .\requirements.txt (line 1)) (0.1.2)
Requirement already satisfied: tzdata in h:\xtts\xtts-webui\venv\lib\site-packages (from tzlocal->dateparser~=1.1.0->gruut[de,es,fr]==2.2.3->tts>=0.22.0->-r .\requirements.txt (line 5)) (2023.4)
Requirement already satisfied: pyasn1<0.6.0,>=0.4.6 in h:\xtts\xtts-webui\venv\lib\site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (0.5.1)
Requirement already satisfied: oauthlib>=3.0.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<2,>=0.5->tensorboard->trainer>=0.0.32->tts>=0.22.0->-r .\requirements.txt (line 5)) (3.2.2)

[notice] A new release of pip available: 22.2.1 -> 23.3.2
[notice] To update, run: python.exe -m pip install --upgrade pip
Looking in indexes: https://download.pytorch.org/whl/cu118
Requirement already satisfied: torch==2.1.1+cu118 in h:\xtts\xtts-webui\venv\lib\site-packages (2.1.1+cu118)
Requirement already satisfied: torchaudio==2.1.1+cu118 in h:\xtts\xtts-webui\venv\lib\site-packages (2.1.1+cu118)
Requirement already satisfied: networkx in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (2.8.8)
Requirement already satisfied: jinja2 in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (3.1.2)
Requirement already satisfied: fsspec in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (2023.12.2)Requirement already satisfied: sympy in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (1.12)
Requirement already satisfied: typing-extensions in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (4.9.0)
Requirement already satisfied: filelock in h:\xtts\xtts-webui\venv\lib\site-packages (from torch==2.1.1+cu118) (3.13.1)
Requirement already satisfied: MarkupSafe>=2.0 in h:\xtts\xtts-webui\venv\lib\site-packages (from jinja2->torch==2.1.1+cu118) (2.1.3)
Requirement already satisfied: mpmath>=0.19 in h:\xtts\xtts-webui\venv\lib\site-packages (from sympy->torch==2.1.1+cu118) (1.3.0)

[notice] A new release of pip available: 22.2.1 -> 23.3.2
[notice] To update, run: python.exe -m pip install --upgrade pip
Install deepspeed for windows for python 3.10.x and CUDA 11.8
Install complete.
Press any key to continue . . .

(venv) H:\XTTS\xtts-webui>start_xtts_webui.bat
2024-01-14 21:14:53.628 | INFO     | xtts_webui:<module>:57 - Start loading model v2.0.2
2024-01-14 21:14:53.628 | INFO     | xtts_webui:<module>:60 - this dir: H:\XTTS\xtts-webui
[2024-01-14 21:15:04,963] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2024-01-14 21:15:05,143] torch.distributed.elastic.multiprocessing.redirects: [WARNING] NOTE: Redirects are currently not supported in Windows or MacOs.
[2024-01-14 21:15:05,325] [INFO] [logging.py:96:log_dist] [Rank -1] DeepSpeed info: version=0.11.2+unknown, git-hash=unknown, git-branch=unknown
[2024-01-14 21:15:05,326] [WARNING] [config_utils.py:69:_process_deprecated_field] Config parameter replace_method is deprecated. This parameter is no longer needed, please remove from your call to DeepSpeed-inference
[2024-01-14 21:15:05,326] [WARNING] [config_utils.py:69:_process_deprecated_field] Config parameter mp_size is deprecated use tensor_parallel.tp_size instead
[2024-01-14 21:15:05,327] [INFO] [logging.py:96:log_dist] [Rank -1] quantize_bits = 8 mlp_extra_grouping = False, quantize_groups = 1
[2024-01-14 21:15:05,520] [INFO] [logging.py:96:log_dist] [Rank -1] DeepSpeed-Inference config: {'layer_id': 0, 'hidden_size': 1024, 'intermediate_size': 4096, 'heads': 16, 'num_hidden_layers': -1, 'dtype': torch.float32, 'pre_layer_norm': True, 'norm_type': <NormType.LayerNorm: 1>, 'local_rank': -1, 'stochastic_mode': False, 'epsilon': 1e-05, 'mp_size': 1, 'scale_attention': True, 'triangular_masking': True, 'local_attention': False, 'window_size': 1, 'rotary_dim': -1, 'rotate_half': False, 'rotate_every_two': True, 'return_tuple': True, 'mlp_after_attn': True, 'mlp_act_func_type': <ActivationFuncType.GELU: 1>, 'specialized_mode': False, 'training_mp_size': 1, 'bigscience_bloom': False, 'max_out_tokens': 1024, 'min_out_tokens': 1, 'scale_attn_by_inverse_layer_idx': False, 'enable_qkv_quantization': False, 'use_mup': False, 'return_single_tuple': False, 'set_empty_params': False, 'transposed_mode': False, 'use_triton': False, 'triton_autotune': False, 'num_kv': -1, 'rope_theta': 10000}
2024-01-14 21:15:05.974 | INFO     | scripts.tts_funcs:load_model:99 - Pre-create latents for all current speakers
2024-01-14 21:15:05.975 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for calm_female: speakers/calm_female.wav
H:\XTTS\xtts-webui\venv\lib\site-packages\torchaudio\functional\functional.py:147: UserWarning: Specified kernel cache directory could not be created! This disables kernel caching. Specified directory is temp/torch/kernels. This warning will appear only once per process. (Triggered internally at ..\aten\src\ATen\native\cuda\jit_utils.cpp:1444.)
  return spec_f.abs().pow(power)
2024-01-14 21:15:07.326 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for davidatt: speakers/davidatt.wav
2024-01-14 21:15:07.456 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for female: speakers/female.wav
2024-01-14 21:15:07.544 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for male: speakers/male.wav
2024-01-14 21:15:07.623 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for MorganFreeman: speakers/MorganFreeman.wav
2024-01-14 21:15:07.721 | INFO     | scripts.tts_funcs:get_or_create_latents:173 - creating latents for NarratorNew: ['speakers/NarratorNew\\reference.wav']
2024-01-14 21:15:07.790 | INFO     | scripts.tts_funcs:create_latents_for_all:187 - Latents created for all 6 speakers.
2024-01-14 21:15:07.790 | INFO     | scripts.tts_funcs:load_model:103 - Model successfully loaded
Running on local URL:  http://127.0.0.1:8010

To create a public link, set `share=True` in `launch()`.
2024-01-14 21:17:32.187 | INFO     | scripts.tts_funcs:unload_model:79 - Model unloaded
Warning, existing language does not match target language. Updated lang.txt with target language.
Loading Whisper Model!
The sum of the duration of the audios that you provided should be at least 2 minutes!
finetuned_models\commentator
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "H:\XTTS\xtts-webui\modules\train.py", line 239, in train_xtts_model
    shutil.copytree(str(ready_folder), str(
  File "C:\Program Files\Python310\lib\shutil.py", line 556, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'finetuned_models\\commentator\\ready'
2024-01-14 21:23:12.341 | INFO     | scripts.tts_funcs:unload_model:79 - Model unloaded
Existing language matches target language
Loading Whisper Model!
Existing training metadata found and loaded.
Existing evaluation metadata found and loaded.
Dataset Processed!
finetuned_models\commentator
>> DVAE weights restored from: H:\XTTS\xtts-webui\models\v2.0.2\dvae.pth
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\modules\train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "H:\XTTS\xtts-webui\scripts\utils\gpt_train.py", line 175, in train_gpt
    train_samples, eval_samples = load_tts_samples(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\TTS\tts\datasets\__init__.py", line 121, in load_tts_samples
    assert len(meta_data_train) > 0, f" [!] No training samples found in {root_path}/{meta_file_train}"
AssertionError:  [!] No training samples found in finetuned_models\commentator\dataset/metadata_train.csv
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "H:\XTTS\xtts-webui\modules\train.py", line 245, in train_xtts_model
    shutil.copy(speaker_reference_path, reference_destination)
  File "C:\Program Files\Python310\lib\shutil.py", line 417, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "C:\Program Files\Python310\lib\shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'finetuned_models\\commentator\\ready\\reference.wav'
2024-01-14 21:24:54.746 | INFO     | scripts.tts_funcs:unload_model:79 - Model unloaded
Existing language matches target language
Loading Whisper Model!
Existing training metadata found and loaded.
Existing evaluation metadata found and loaded.
Dataset Processed!
finetuned_models\commentator
>> DVAE weights restored from: H:\XTTS\xtts-webui\models\v2.0.2\dvae.pth
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\modules\train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "H:\XTTS\xtts-webui\scripts\utils\gpt_train.py", line 175, in train_gpt
    train_samples, eval_samples = load_tts_samples(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\TTS\tts\datasets\__init__.py", line 121, in load_tts_samples
    assert len(meta_data_train) > 0, f" [!] No training samples found in {root_path}/{meta_file_train}"
AssertionError:  [!] No training samples found in finetuned_models\commentator\dataset/metadata_train.csv
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 2134, in run_sync_in_worker_thread
    return await future
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 851, in run
    result = context.run(func, *args)
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\gradio\utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "H:\XTTS\xtts-webui\modules\train.py", line 245, in train_xtts_model
    shutil.copy(speaker_reference_path, reference_destination)
  File "C:\Program Files\Python310\lib\shutil.py", line 417, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "C:\Program Files\Python310\lib\shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'finetuned_models\\commentator\\ready\\reference.wav'
2024-01-14 21:28:26.131 | INFO     | scripts.tts_funcs:unload_model:79 - Model unloaded
Existing language matches target language
Loading Whisper Model!
Existing training metadata found and loaded.
Existing evaluation metadata found and loaded.
Dataset Processed!
finetuned_models\commentator
>> DVAE weights restored from: H:\XTTS\xtts-webui\models\v2.0.2\dvae.pth
Traceback (most recent call last):
  File "H:\XTTS\xtts-webui\modules\train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "H:\XTTS\xtts-webui\scripts\utils\gpt_train.py", line 175, in train_gpt
    train_samples, eval_samples = load_tts_samples(
  File "H:\XTTS\xtts-webui\venv\lib\site-packages\TTS\tts\datasets\__init__.py", line 121, in load_tts_samples
    assert len(meta_data_train) > 0, f" [!] No training samples found in {root_path}/{meta_file_train}"
AssertionError:  [!] No training samples found in finetuned_models\commentator\dataset/metadata_train.csv

```

**Comment by daswer123** [comment]
Hi, sorry for not replying earlier, lots of work going on right now.

Try deleting the model in the finetune_models folder , and try again

**Comment by GamingDaveUk** [comment]
> Hi, sorry for not replying earlier, lots of work going on right now.
> 
> Try deleting the model in the finetune_models folder , and try again

NP You have a life chap, no one expects immediate replies :)

I do have bad news though.

I created a new folder and started afresh. Same issues popped up.

**Comment by GamingDaveUk** [comment]
> Hi, sorry for not replying earlier, lots of work going on right now.
> 
> Try deleting the model in the finetune_models folder , and try again

Just tried again with another fresh install as i see there was a commit 6 hours ago, but the same issues remain. Can I ask have you tried to recreate the steps I am doing on a fresh install your end?

So I load the webui,
Click on train
In finetune model name I write commentator
I drop my audio sample ont the audio input section and wait for it to upload.
I click train.

I cant see why it wouldnt work on my system if its working on yours but i have even tried installing it on a different hard drive and getting the exact same error sequence.

**Comment by GamingDaveUk** [comment]
> Hi, sorry for not replying earlier, lots of work going on right now.
> 
> Try deleting the model in the finetune_models folder , and try again

While I stand by what i said about not needing instant replies, the issue is still ongoing for 4 days now, is there a fix in the works?

**Comment by GamingDaveUk** [comment]
Ok
The issue (as explained to me on discord by MattiaHeron ) is that the samples i am using are not getting detected and processed by whisper (despite the fact the same samples were before a few updates) and so there are no samples going into the dataset and the app gives the error.
Since I am using one large sample it means if it fails thats it, no training dataset.
The chap sent me some test samples where he split them up manually into tracks and that worked (since if one fails there is still tons). 
I am going to leave this as a bug though since files that used to work now dont. but atleast now we know why its erroring and hopefully a more verbose error message could be added "No samples processed please upload better quality samples" or some such. 
Huge thankyou to MattiaHeron who helped problem solve the issue when I was looking for an alternative to the WebUI.

**Comment by zhuochengwang** [comment]
Same issue.

`Existing language matches target language
Loading Whisper Model!
Existing training metadata found and loaded.
Existing evaluation metadata found and loaded.
Segments from Voice123 - Jose Araujo Cardiff Castle audio tour - Jose Araujo have been previously processed; skipping...
Dataset Processed!
finetuned_models/jose_pteu_base_model
>> DVAE weights restored from: /content/xtts-webui/models/v2.0.2/dvae.pth
 | > Found 29 files in /content/xtts-webui/finetuned_models/jose_pteu_base_model/dataset
usage: app.py [-h] [--continue_path CONTINUE_PATH] [--restore_path RESTORE_PATH]
              [--best_path BEST_PATH] [--use_ddp true/false] [--use_accelerate true/false]
              [--grad_accum_steps GRAD_ACCUM_STEPS] [--overfit_batch true/false]
              [--skip_train_epoch true/false] [--start_with_eval true/false]
              [--small_run SMALL_RUN] [--gpu GPU] [--rank RANK] [--group_id GROUP_ID]
app.py: error: argument -h/--help: ignored explicit argument 's'
Traceback (most recent call last):
  File "/usr/lib/python3.10/argparse.py", line 1878, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib/python3.10/argparse.py", line 2091, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib/python3.10/argparse.py", line 1999, in consume_optional
    raise ArgumentError(action, msg % explicit_arg)
argparse.ArgumentError: argument -h/--help: ignored explicit argument 's'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/content/xtts-webui/modules/train.py", line 87, in train_model
    speaker_xtts_path, config_path, original_xtts_checkpoint, vocab_file, exp_path, speaker_wav = train_gpt(
  File "/content/xtts-webui/scripts/utils/gpt_train.py", line 183, in train_gpt
    trainer = Trainer(
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 397, in __init__
    args, coqpit_overrides = self.parse_argv(args)
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 680, in parse_argv
    training_args, coqpit_overrides = parser.parse_known_args()
  File "/usr/lib/python3.10/argparse.py", line 1881, in parse_known_args
    self.error(str(err))
  File "/usr/lib/python3.10/argparse.py", line 2606, in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
  File "/usr/lib/python3.10/argparse.py", line 2593, in exit
    _sys.exit(status)
SystemExit: 2
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/gradio/queueing.py", line 489, in call_prediction
    output = await route_utils.call_process_api(
  File "/usr/local/lib/python3.10/dist-packages/gradio/route_utils.py", line 232, in call_process_api
    output = await app.get_blocks().process_api(
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1561, in process_api
    result = await self.call_function(
  File "/usr/local/lib/python3.10/dist-packages/gradio/blocks.py", line 1179, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/usr/local/lib/python3.10/dist-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/usr/local/lib/python3.10/dist-packages/gradio/utils.py", line 678, in wrapper
    response = f(*args, **kwargs)
  File "/content/xtts-webui/modules/train.py", line 245, in train_xtts_model
    shutil.copy(speaker_reference_path, reference_destination)
  File "/usr/lib/python3.10/shutil.py", line 417, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.10/shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'finetuned_models/jose_pteu_base_model/ready/reference.wav'`

**Comment by Snobbias** [comment]
> Ok The issue (as explained to me on discord by MattiaHeron ) is that the samples i am using are not getting detected and processed by whisper (despite the fact the same samples were before a few updates) and so there are no samples going into the dataset and the app gives the error. Since I am using one large sample it means if it fails thats it, no training dataset. The chap sent me some test samples where he split them up manually into tracks and that worked (since if one fails there is still tons). I am going to leave this as a bug though since files that used to work now dont. but atleast now we know why its erroring and hopefully a more verbose error message could be added "No samples processed please upload better quality samples" or some such. Huge thankyou to MattiaHeron who helped problem solve the issue when I was looking for an alternative to the WebUI.

Thanks mate. You saved me from even more troubleshooting. Been trying to get this to work all day. Will make it a habit to split audio files into many smaller files. Something is probably still not right since GPU usage is never above 20% during training, but that will be explored another day.

**Comment by 78Alpha** [comment]
Adding on to this, I loaded 4909 small files and the same issue occurred. Now, this is a fresh install on a different drive because I didn't want my SSD to be thrashed. I pre-split them with a StyleTTS2 tool as it happened to match all the boxes that XTTS needs in audio. Sadly, it did not workout and does not explain anything other than the above issues. I would hope to have seen a "no audio files" or "failed to create dataset" sort of error.
